dmutils/logging.py

```diff
@@ -120,6 +123,25 @@ def filter(self, record):
         return record
 
 
+class CustomLogFormatter(logging.Formatter):
+    """Accepts a format string for the message and formats it with the extra fields"""
+
+    FORMAT_STRING_FIELDS_PATTERN = re.compile(r'\((.+?)\)', re.IGNORECASE)
```

Comment:
I'm not sure this needs the extra brackets - i.e. I think `\(.+?\)` will do the same thing?

Do we have to use a regex here - having a single string list of things in brackets seems weird, but is that a constraint of the Formatter?